### ISSUE TITLE
Compose key mapping not taken from kde plasma/kwin wayland (as virtual keyboard)

### ISSUE THREAD

**Issue by pdvrieze** [ibus/ibus#2579]
Please fill in the following items if you don't know the root cause.

Which distribution and version?: 
Gentoo

Which desktop environment and version?: 
Plasma 5.27.9

Which session type?: 
Wayland

Which application and version?: 
Any client

IBus version?: 
ibus-1.5.29_rc1

Issue description:
When running as input method the system keyboard is not inherited. This means that the compose key remapping from the "compose:menu" option is not active in the input method. While many layouts are available they do not support libxkbcommon options.

I'd like some way to have compose key (Multi_Key) supported. Probably the easiest is to support inheriting the system layout on wayland. Adding support for options would be good too.

Steps to reproduce:
1. Add "xkb:us::eng" engine
2. Add "xkb:gb::eng" engine
3. Enable "Use system keyboard layout" (it also doesn't work with it disabled)
4. Note that both layouts work/switch
5. Note also that there is no place to specify keyboard options (textual or graphical configuration)
6. When switching to not having a virtual keyboard compose works correctly

Can you reproduce your problem when you restart ibus-daemon? (yes / no): yes

Do you see any errors when you run ibus-daemon with the verbose option?:
(Run `ibus-daemon --xim --verbose &` and look at the output when you encounter your problem.)

Can you reproduce your problem with a new user account instead of the current your account? (yes / no): 
This is a missing feature

**Comment by fujiwarat** [comment]
How do you run ibus-daemon?

> 6. When switching to not having a virtual keyboard compose works correctly

As I explained the setup in https://groups.google.com/g/ibus-user/c/iXaBdVm_lV0, IBus is enabled as a virtual keyboard so I think you cannot specify another virtual keyboard.

I think you can enable compose key with systemsettings5 -> "Input Devices" -> "Keyboard" -> "Advanced".

**Comment by pdvrieze** [comment]
@fujiwarat "Switching to not having a virtual keyboard" means that I use the default input system without using ibus. I enabled compose through the advanced settings, the problem is that ibus (as virtual keyboard) doesn't observe that setting (even when "use system keyboard" is enabled) - it looks as though the keyboard mappings are as specified in the ibus settings, rather than the system ones.

Note that when running ibus as independent daemon with qt/gtk immodule it does work correctly, but at a different layer.

**Comment by fujiwarat** [comment]
You're right. I checked your issue with `xev` command by mistake while IBus does not support XIM in Plasma Wayland yet.
I understand your issue now and I will refer how GNOME Wayland handle XKB and compose keys.

**Comment by pdvrieze** [comment]
@fujiwarat The problem is basically not supporting the mapping given by the option (there is no physical compose key, but I want to use one of the options in xkb to pretend that menu is a compose key). (As the engines currently don't support options, only variants) - it also seems as though the list of engines is somehow hardcoded - but I didn't dive too deep into the code.
